presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/client/ElasticsearchClient.java

```diff
@@ -184,6 +189,15 @@ private static RestHighLevelClient createClient(ElasticsearchConfig config, Opti
                 .setMaxRetryTimeoutMillis((int) config.getMaxRetryTime().toMillis());
 
         builder.setHttpClientConfigCallback(clientBuilder -> {
+            if (config.getSecurity().isPresent()) {
```

Comment:
Once you move the configs to a separate object similar to `AwsSecurityConfig`, we'll need an additional `Optional<XPackSecurityConfig>` argument to this method and this check will need to be adjusted.